### ISSUE TITLE
MongoError: database name must be a string

### ISSUE THREAD

**Issue by alias-mac** [jdesboeufs/connect-mongo#335]
Based on the code here:
https://github.com/jdesboeufs/connect-mongo/blob/master/src/index.js#L140

It seems we aren't able to provide a DB name, although all tutorials make the connection with the URL separate from the db name.
http://mongodb.github.io/node-mongodb-native/3.2/tutorials/connect/

```js
const MongoClient = require('mongodb').MongoClient;
const assert = require('assert');

// Connection URL
const url = 'mongodb://localhost:27017';

// Database Name
const dbName = 'myproject';

// Create a new MongoClient
const client = new MongoClient(url);

// Use connect method to connect to the Server
client.connect(function(err) {
  assert.equal(null, err);
  console.log("Connected successfully to server");

  const db = client.db(dbName);

  client.close();
});
```

Using:
```js
app.use(session({
  store: new MongoStore({ client: client })
}));
```
Fails with `MongoError: database name must be a string`. Instead we should be able to pass a DB name or passing the db directly like so:
```js
app.use(session({
  store: new MongoStore({ db })
}));
```

It seems that typescript definition is already pointing that you will support a `db` option tho, but I don't see it in this repository:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/connect-mongo/index.d.ts#L95

I'm willing to make a PR if you tell me which approach you prefer.

**Comment by AlainEdwards** [comment]
Hey @alias-mac,

You are correct about this problem. I am experiencing the same problem and was about to create an issue ticket for it. `client.db()` does not return an already existent dbo. The only way I was able to bypass this problem while still using a Mongo Client object was to write a prototype `handleNewConnectionAsync` function.

**Comment by byeokim** [comment]
https://github.com/jdesboeufs/connect-mongo/blob/44f1850943eebadd7de3e4dbdc2a513c93e57cf0/src/index.js#L138-L144

My workaround is to put database name in the MongoDB [connection string](https://docs.mongodb.com/manual/reference/connection-string/) because `client.db()` would look into connection string if database name was not provided as argument: [[1]](https://github.com/mongodb/node-mongodb-native/blob/0463f7838dbfc0740f6fa52174b96241eb0c3980/lib/mongo_client.js#L250-L256)

```javascript
// https://github.com/mongodb/node-mongodb-native/lib/mongo_client.js

MongoClient.prototype.db = function(dbName, options) {
  options = options || {};


  // Default to db from connection string if not provided
  if (!dbName) {
    dbName = this.s.options.dbName;
  }
```

But there is no fallback in case when database name is not present in connection string too. So my suggestion is to have a default database name, in addition to @alias-mac's suggestion which is to be able to pass a database name.

**Comment by mingchuno** [comment]
@byeokim Yes! I would love a PR on it 🙇

**Comment by byeokim** [comment]
I found that default database name `test` was [added to `mongodb` module](https://github.com/mongodb/node-mongodb-native/blob/master/lib/core/uri_parser.js#L622-L625). So we would no longer encounter `MongoError: database name must be a string`. However we may now see `MongoError: not authorized on test to execute command` if we can't provide database name because `mongodb` will set database name to `test`. So I created PR #343 to address this issue.
